src/charmed_kubeflow_chisme/components/pebble_component.py

```diff
@@ -218,7 +218,9 @@ def __init__(
         # TODO: Should a PebbleComponent automatically be subscribed to this event?  Or just
         #  a PebbleServiceComponent?
         self._events_to_observe: List[str] = [
-            get_pebble_ready_event_from_charm(self._charm, self.container_name)
+            get_event_from_charm(self._charm, self.container_name, "pebble_ready"),
+            get_event_from_charm(self._charm, self.container_name, "pebble_check_failed"),
+            get_event_from_charm(self._charm, self.container_name, "pebble_check_recovered"),
```

Comment:
question: how does this behave when a pebble service has more than one check? Is each of these events tied to all checks?

Reply:
Yes! Both events are emitted whenever a single check fails, and we can get the check's name via `event.str`. see the class's [documentation](https://ops.readthedocs.io/en/latest/index.html#ops.PebbleCheckFailedEvent)

Reply:
SGTM for multiple checks. I'm wondering also if the charm has no checks? would the `charm.on.pebble_check_failed` still exist? to make sure that `getattr` will not raise `AttributeError`.

Reply:
Yes, the attribute always exists, I checked on a local deployment